### DISCUSSION TITLE
Create internal lib

### DISCUSSION THREAD

**Discussion by gabryellr** [TNG/ArchUnit#1213] [Q&A]
Hi, I need help, I'm going to create a library for my team, for example: timeA-archunit will be the name of the library, in this lib we will implement archunit and create the tests, then we will add this lib to our projects, looking like this:

projectA will have a dependency on our timeA-archunit lib, because when we need to add more validations, we just add them to our lib and bump the version in the other services that use our timeA-archunit, how can I achieve it?

**Comment by hankem** [comment]
If you bundle all tests in a single class `MyArchRuleSuite` in `timeA-archunit` (or a few which do not change much), you could use 
```
@ArchTest
static final ArchTests includedRules = ArchTests.in(MyArchRuleSuite.class);
```
in case you're using [JUnit 5](https://www.javadoc.io/doc/com.tngtech.archunit/archunit-junit5-api/latest/com/tngtech/archunit/junit/ArchTests.html) or [JUnit 4](https://www.javadoc.io/doc/com.tngtech.archunit/archunit-junit4/latest/com/tngtech/archunit/junit/ArchTests.html).

**Comment by hankem** [comment]
You can probably also use the [`arch-unit-gradle-plugin`](https://github.com/societe-generale/arch-unit-gradle-plugin) or [`arch-unit-maven-plugin`](https://github.com/societe-generale/arch-unit-maven-plugin) in case you're using Gradle or Maven, respectively.
